src/init.js

```diff
@@ -407,6 +411,41 @@ export const init = async (options) => {
 	}
 };
 
+export const setTagProtection = async function (options, details) {
+	const pattern = '*';
+	const { user } = options;
+
+	let octokit = new Octokit({
+		auth: user.token,
+	});
+
+	const { organization, name } = details;
+
+	if (!options.dev && organization && name) {
+		console.log(`Setting tag protection for ${organization}/${name}...`);
+
+		try {
+			// create tag protection rule repository
+			const tagProtectionResponse = await octokit.rest.repos.createTagProtection({
+				owner: organization,
+				repo: name,
+				pattern,
+			});
+
+			if (tagProtectionResponse?.status === 200 || tagProtectionResponse?.status === 201) {
+				console.log(chalk.green(`created tag protection rule '${pattern}'`));
+			} else {
+				throw new Error(err);
```

Comment:
I think `err` would be undefined here as it's outside the catch below?

Reply:
Good catch!